src/Registration/cReg/NiftiImageData.cpp

```diff
@@ -1742,11 +1742,34 @@ void NiftiImageData<dataType>::dot(const DataContainer& a_x, void* ptr) const
 {
     const NiftiImageData<dataType>& x = dynamic_cast<const NiftiImageData<dataType>&>(a_x);
     ASSERT(_nifti_image->nvox == x._nifti_image->nvox, "dot operands size mismatch");
-    double s = 0.0;
-    for (unsigned i=0; i<this->_nifti_image->nvox; ++i)
-        s += double(_data[i] * x._data[i]);
+    float s = 0.0;
```

Comment:
not sure why you make this `float` now. Will lead to loss of precision.

Also wouldn't handle complex data type (but `dot` then probably needs a conjugate anyway, so I guess it might not be used then?)